lessons/module-3/unit-testing-react.md

```diff
@@ -461,17 +466,33 @@ it('should call the onPurchase prop when clicked', () => {
 
 #### Your Turn
 
-- Can you add an `onClick` function to the "Purchase" button and be the hero who makes 
-  the test pass?
+- To the Grocery component, add an `onClick` function to a "Purchase" button and make the test pass.
 - We likely want to pass in a grocery ID or grocery name to the `onPurchase` method so 
-  we can keep track of what has been purchased. Can you add an assertion to the previous 
-  test to check that `onPurchaseMock` was called with the correct arguments? 
+  we can keep track of what has been purchased. 
+- Add an assertion to the previous test to check that `onPurchaseMock` was called with the correct arguments. 
   ([Hint](https://facebook.github.io/jest/docs/expect.html#content))
-- Can you write the tests and implementation for the "Star" and "Remove" buttons?
+- Write the tests and implementation for the "Star" and "Remove" buttons.
+
+- Here is some code to help get you started: 
+```
+const Grocery = ({ name, quantity, notes, purchased, starred, onPurchase, onStar, onDelete }) => {
```

Comment:
This code should not be here. The point of doing this in a test driven way is to write the tests that allow for creating this component.

Reply:
Yes, I would like to see the lesson plan advance the student learning and acceptance of TDD.  This is the student's first deep lesson on TDD in the Mod.  I can only reference my own experience - for me, having some of this code available so I could study the test and the code it tested is very useful in making the connections of what specifically tests are isolating in the code.  The mod has many opportunities to practice TDD when some basic connections have been established.